### ISSUE TITLE
mini.statusline: Status is not drawn on setup()

### ISSUE THREAD

**Issue by UtkarshVerma** [echasnovski/mini.nvim#728]
### Contributing guidelines

- [X] I have read [CONTRIBUTING.md](https://github.com/echasnovski/mini.nvim/blob/main/CONTRIBUTING.md)
- [X] I have read [CODE_OF_CONDUCT.md](https://github.com/echasnovski/mini.nvim/blob/main/CODE_OF_CONDUCT.md)
- [X] I have updated 'mini.nvim' to latest version

### Module(s)

mini.statusline

### Description

When we call `require("mini.statusline").setup(opts)`, the plugin is loaded, but the status is not drawn. This causes issues when I try to lazy load using `lazy.nvim` which loads the plugin after `WinEnter`. To get it to draw the initial status, I have to add this after the setup as a hack.

```lua
vim.cmd([[do WinEnter]])
```

### Neovim version

0.9.5

### Steps to reproduce

1. `nvim -nu minimal.lua`

```lua
-- minimal.lua
-- DO NOT change the paths and don't remove the colorscheme
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "cache" }) do
	vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
	vim.fn.system({ "git", "clone", "--filter=blob:none", "https://github.com/folke/lazy.nvim.git", lazypath })
end
vim.opt.runtimepath:prepend(lazypath)

local config = require("config")
vim.opt.laststatus = 2

require("lazy").setup({
	{
		"catppuccin/nvim",
		name = "catppuccin",
		lazy = false,
		priority = 1000,
		opts = {
			integrations = {
				mini = true,
			},
		},
		config = function(_, opts)
			require("catppuccin").setup(opts)
			vim.cmd.colorscheme("catppuccin")
		end,
	},
	{
		"echasnovski/mini.statusline",
		event = "VeryLazy",
		opts = {
			-- Do not override vim settings.
			set_vim_settings = false,
		},
	},
}, {
	root = root .. "/plugins",
	defaults = {
		lazy = true,
		version = "*",
	},
	install = {
		colorscheme = { "catppuccin" },
	},
})
```

### Expected behavior

The statusline is rendered on startup.

### Actual behavior

The statusline is not rendered, instead nvim's default status is shown.

![image](https://github.com/echasnovski/mini.nvim/assets/31820255/ed50fe43-5b1c-4d5a-897e-c66bfa4da7a9)

**Comment by echasnovski** [comment]
Thanks for the issue!

There are several things to unpack here:
- First, I'd **strongly** suggest to not lazy load any UI related plugin (that is, which is expected to be active right after Neovim has started).
- Second, reproducible example contains code which is not executable in fresh setup: `local config = require("config")`.
- Third, reproducible steps do not use latest 'mini.nvim' version (that is, `main` branch), but latest stable release (as indicated by `version = "*"`). Coincidentally, this issue was solved [two days ago](47b0e0eb9ad70e5bab10f48281e574a68d71f511) just after 0.12.0 stable release.

So I'd suggest to either not lazy load 'mini.statusline' (use `lazy = false` similar to how it is done in 'catppuccin/nvim') or use latest `main` (add `version = false` to 'mini.statusline' config).

Closing as already solved.

**Comment by UtkarshVerma** [comment]
Thanks, it works.
